components/bl-mapbox-component/src/map.js

```diff
@@ -1,27 +1,26 @@
-import { useEffect, useRef, useState } from 'react';
+import { useEffect, useRef } from 'react';
 
 import mapboxgl from './lib/mapbox';
-import { initMapboxLibrary, useMarkers, usePolygons } from './helpers';
+import { initMapboxLibrary, useMarkers, usePolygons, Map } from './helpers';
 
 const { cn } = BackendlessUI.CSSUtils;
 
-export default function Map({ component, eventHandlers, settings }) {
+export default function Mapbox({ component, eventHandlers, settings }) {
   const mapContainerRef = useRef(null);
   const mapRef = useRef(null);
 
-  const [polygonsArray, setPolygonsArray] = useState([]);
-  const [markersArray, setMarkersArray] = useState([]);
-
   const { accessToken } = settings;
 
   const { onMarkerClick, onPolygonClick } = eventHandlers;
 
   const { markers, polygons, center, classList } = component;
 
+  const map = new Map(mapRef);
```

Comment:
it creates a new instance on each render and then in the next useEffect it remembers the first one, which means the `initMapboxLibrary ` runs only for the first instance